src/test/java/TasksServletTest.java

```diff
@@ -112,48 +143,109 @@ public void noTaskLists() throws IOException, ServletException {
+    Mockito.when(tasksClient.listTasks(TASKLIST_ONE)).thenReturn(TASKS_COMPLETED_YESTERDAY);
+    servlet.doGet(request, response);
+    printWriter.flush();
+    TasksResponse tasksResponse = gson.fromJson(stringWriter.toString(), TasksResponse.class);
+    Assert.assertEquals(ONE_TASKLIST_TITLES, tasksResponse.getTaskListNames());
+    Assert.assertEquals(0, tasksResponse.getTasksToComplete());
+    Assert.assertEquals(0, tasksResponse.getTasksDueToday());
+    Assert.assertEquals(0, tasksResponse.getTasksCompletedToday());
+    Assert.assertEquals(0, tasksResponse.getTasksOverdue());
+  }
+
+  @Test
+  public void taskCompletedToday() throws IOException, ServletException {
+    Mockito.when(tasksClient.listTaskLists()).thenReturn(ONE_TASKLIST);
+    Mockito.when(tasksClient.listTasks(TASKLIST_ONE)).thenReturn(TASKS_COMPLETED_TODAY);
     servlet.doGet(request, response);
     printWriter.flush();
-    List<Task> tasks = gson.fromJson(stringWriter.toString(), LIST_OF_TASKS_TYPE);
-    Assert.assertEquals(2, tasks.size());
-    Assert.assertThat(tasks, CoreMatchers.hasItems(TASKS_ONE_TWO.get(0), TASKS_ONE_TWO.get(1)));
+    TasksResponse tasksResponse = gson.fromJson(stringWriter.toString(), TasksResponse.class);
+    Assert.assertEquals(ONE_TASKLIST_TITLES, tasksResponse.getTaskListNames());
+    Assert.assertEquals(0, tasksResponse.getTasksToComplete());
+    Assert.assertEquals(0, tasksResponse.getTasksDueToday());
+    Assert.assertEquals(1, tasksResponse.getTasksCompletedToday());
+    Assert.assertEquals(0, tasksResponse.getTasksOverdue());
   }
 
   @Test
-  public void completeTaskList() throws IOException, ServletException {
-    Mockito.when(tasksClient.listTaskLists()).thenReturn(SOME_TASKLISTS);
-    Mockito.when(tasksClient.listTasks(TASKLIST_ONE)).thenReturn(TASKS_ONE_TWO);
-    Mockito.when(tasksClient.listTasks(TASKLIST_TWO)).thenReturn(TASKS_THREE_FOUR);
+  public void allTasks() throws IOException, ServletException {
+    Mockito.when(tasksClient.listTaskLists()).thenReturn(ONE_TASKLIST);
+    Mockito.when(tasksClient.listTasks(TASKLIST_ONE)).thenReturn(ALL_TASKS);
     servlet.doGet(request, response);
     printWriter.flush();
-    List<Task> tasks = gson.fromJson(stringWriter.toString(), LIST_OF_TASKS_TYPE);
-    Assert.assertEquals(4, tasks.size());
-    Assert.assertThat(
-        tasks,
-        CoreMatchers.hasItems(
-            TASKS_ONE_TWO.get(0),
-            TASKS_ONE_TWO.get(1),
-            TASKS_THREE_FOUR.get(0),
-            TASKS_THREE_FOUR.get(1)));
+    TasksResponse tasksResponse = gson.fromJson(stringWriter.toString(), TasksResponse.class);
+    Assert.assertEquals(ONE_TASKLIST_TITLES, tasksResponse.getTaskListNames());
+    Assert.assertEquals(3, tasksResponse.getTasksToComplete());
+    Assert.assertEquals(1, tasksResponse.getTasksDueToday());
+    Assert.assertEquals(1, tasksResponse.getTasksCompletedToday());
+    Assert.assertEquals(1, tasksResponse.getTasksOverdue());
   }
 }
```

Comment:
You should add a test with more than one task list.

Reply:
Added, thanks!

Reply:
`RESOLVED` https://github.com/googleinterns/step7-2020/pull/105